master/buildbot/worker/base.py

```diff
@@ -77,6 +79,8 @@ def checkConfig(self, name, password, max_builds=None,
         @type locks: dictionary
         """
         self.name = name = ascii2unicode(name)
+        self.build_wait_timeout = build_wait_timeout
```

Comment:
indeed, there are still a bunch of services that needs to be migrating to the reconfigure api.
schedulers and workers are one of those. 
This means buildbot reload won't apply changes for those.

I need to make the list, and see if we want to release with our without the fix.
not complicated, but requires time to make sure this is properly tested.